enzyme/Enzyme/GradientUtils.h

```diff
@@ -2117,7 +2117,7 @@ class DiffeGradientUtils : public GradientUtils {
         idx2.push_back(v);
         auto selects = addToDiffe(
             val, BuilderM.CreateExtractValue(dif, ArrayRef<unsigned>(i)),
-            BuilderM, nullptr, idx2);
+            BuilderM, addingType, idx2);
```

Comment:
Same comment

Reply:
E.g. I would've expected addingType to also be an array and needing to extract the inner type as the inner addingType

Reply:
That said usually the result here comes directly from TA and that probably gives a scalar -- but this is likely why this was nullptr.